src/foundry/common/utils/helpers.mjs.d.ts

```diff
@@ -172,6 +179,7 @@ export declare function encodeURL(path: string): string;
  *              (default: `0`)
  * @returns An expanded object
  */
+export declare function expandObject<T extends Record<string, unknown>>(obj: FlatObject<T>, _d?: number): T;
```

Comment:
The inference here isn't going to be ideal here, I think, based upon my prior experience with "inverse inference" -- where to infer `T` you have to take what `obj` is and apply the inverse of `FlatObject` which is obviously `ExpandObject` here. While on paper it's something the compiler could attempt to do, `T` is probably going to be inferred overly broad as like `Record<string, unknown>` or something. Here's some test cases you can add to try it out:
```ts
expectType<{ foo: { bar: 1 } }>(expandObject({ "foo.bar": 1 }));
expectType<{ foo: { bar: 1 } }>(expandObject({ "foo.bar": 1, "foo.lorem": 2 }));
expectType<{ foo: { bar: 1, lorem: 2 } }>(expandObject({ "foo": { "bar": 1 }, "foo.lorem": "2" }));
```
We can get this draft in first and I can add in an `ExpandObject` helper type later if so desired as this is strictly an improvement.